src/example-types/date.js

```diff
@@ -1,39 +1,87 @@
+let _ = require('lodash/fp')
 let F = require('futil')
-let moment = require('moment')
+let moment = require('moment-timezone')
 let datemath = require('@elastic/datemath')
 
+let getStartOfQuarter = (quarterOffset, timezone) => {
+  let quarter =
+    moment()
+      .tz(timezone)
+      .quarter() + quarterOffset
+  return moment()
+    .tz(timezone)
+    .quarter(quarter)
+    .startOf('quarter')
+}
+
+let getEndOfQuarter = date =>
+  moment(date)
+    .add(1, 'Q')
+    .subtract(1, 'ms')
+
+let quarterToOffset = { thisQuarter: 0, lastQuarter: -1, nextQuarter: 1 }
+
+// https://www.elastic.co/guide/en/elasticsearch/reference/7.x/common-options.html#date-math
+let rangeToDatemath = {
+  last3Days: { from: 'now-3d', to: 'now' },
+  last7Days: { from: 'now-7d', to: 'now' },
+  last30Days: { from: 'now-30d', to: 'now' },
+  last90Days: { from: 'now-90d', to: 'now' },
+  last180Days: { from: 'now-180d', to: 'now' },
+  last12Months: { from: 'now/d-12M', to: 'now' },
+  last15Months: { from: 'now/d-15M', to: 'now' },
+  last18Months: { from: 'now/d-18M', to: 'now' },
+  last24Months: { from: 'now/d-24M', to: 'now' },
+  last36Months: { from: 'now/d-36M', to: 'now' },
+  last48Months: { from: 'now/d-48M', to: 'now' },
+  last60Months: { from: 'now/d-60M', to: 'now' },
+  lastCalendarMonth: { from: 'now-1M/M', to: 'now/M-1ms' },
+  lastCalendarYear: { from: 'now-1y/y', to: 'now/y-1ms' },
+  thisCalendarMonth: { from: 'now/M', to: 'now+1M/M-1ms' },
+  thisCalendarYear: { from: 'now/y', to: 'now' },
```

Comment:
should this be `-1ms` like this calendar month?

Reply:
This should probably be `now+1y/y-1ms` to be consistent with `thisCalendarMonth`. Thoughts?

Reply:
I changed it to the above.